### PR TITLE
Library update

### PR DESCRIPTION
**Pull request by vivek-devtron**
# Description
Library updates

Fixes # (issue)
https://github.com/devtron-labs/dashboard/issues/327

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/app/details/appDetails/AppDetails.tsx [hunk 1]
```diff
@@ -163,11 +163,11 @@ export const Details: React.FC<{
...
     const [pollingIntervalID, setPollingIntervalID] = useState(null);
-    let prefix = '';
-    if (process.env.NODE_ENV === 'production') {
+    //let prefix = '';
+    //if (process.env.NODE_ENV === 'production') {
         //     //@ts-ignore
-        //     prefix = `${location.protocol}//${location.host}`; 
-    }
+        //     prefix = `${location.protocol}//${location.host}`;
+    //}
```
Comment (sohel-rp) [inline src/components/app/details/appDetails/AppDetails.tsx:170]: Please remove if not in use

File: src/components/charts/list/DiscoverCharts.tsx [hunk 2]
```diff
@@ -412,8 +412,8 @@ export default function DiscoverCharts() {
         <Route path={`${path}/group`}>
             <ChartGroupList />
         </Route>
-        <Route path={`${path}/chart/:chartId/chart-value/:chartValueId?`} render={(props) => {
-            return <ChartValues location={props.location} match={props.match} history={props.history} />
+        <Route path={`${path}/chart/:chartId/chart-value/:chartValueId?`} render={({location, history, match}: {location: any, history: any, match: any}) => {
```
Comment (sohel-rp) [inline src/components/charts/list/DiscoverCharts.tsx:415]: If possible, can we give specific types instead of any?

**Review by sohel-rp — APPROVED** [review]

File: src/components/app/details/appDetails/AppDetails.tsx (hunk 1 above)
Comment (shivani170) [reply src/components/app/details/appDetails/AppDetails.tsx:170]: any reason, not to remove it??

File: src/components/notifications/AddNotification.tsx [hunk 3]
```diff
@@ -50,7 +50,7 @@ interface AddNotificationState {
     sesConfigOptions: { id: number; configName: string; dest: "slack" | "ses" | ""; recipient: string }[];
     isLoading: boolean;
     appliedFilters: Array<{ type: string, value: number | string | undefined, label: string | undefined }>;
-    selectedChannels: { __isNew__: boolean; label: string; value; data: { dest: "slack" | "ses" | "", configId: number; recipient: string; } }[];
+    selectedChannels: { __isNew__?: boolean; label: string; value; data: { dest: "slack" | "ses" | "", configId: number; recipient: string; } }[];
```
Comment (shivani170) [inline src/components/notifications/AddNotification.tsx:53]: type of value id missing

File: src/components/workflowEditor/workflowEditor.tsx [hunk 4]
```diff
@@ -285,14 +285,14 @@ class WorkflowEdit extends Component<WorkflowEditProps, WorkflowEditState> {
                 />
                 <Route
                     path={`${this.props.match.path}/linked-ci/:ciPipelineId`}
-                    render={(props) => {
-                        let ciNode = this.state.allCINodeMap.get(props.match.params.ciPipelineId);
+                    render={({location, history, match}: {location: any, history: any, match: any}) => {
+                        let ciNode = this.state.allCINodeMap.get(match.params.ciPipelineId);
```
Comment (shivani170) [inline src/components/workflowEditor/workflowEditor.tsx:289]: Why are we using location and history type as any and not explicitly ?

**Review by shivani170 — APPROVED** [review]

**Pull request merged**